[OKTA-460124] "Protect Your API" PHP Quickstart (#3147)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/protect-your-api/main/php/reqautheverything.md

> First, create a function that will check whether a request has a valid access token. This function will be responsible for parsing the incoming `Authorization` header and validating the JWT.

Author: fixed

```php
function hasValidAccessToken() {
  // Require an access token is sent in the HTTP Authorization header
  if(!isset($_SERVER['HTTP_AUTHORIZATION'])) {
    return false;
  }

  $accessToken = explode(' ', $_SERVER['HTTP_AUTHORIZATION'])[1];

  $keys = getJWKS();

  try {
    $jwt = \Jose\Easy\Load::jws($accessToken)
      ->algs(['RS256'])
      ->keyset($keys)
      ->exp()
      ->iat()
      ->iss($_ENV['OKTA_OAUTH2_ISSUER'])
      ->aud($_ENV['OKTA_AUDIENCE'])
      ->run();
      ;
  } catch(\Exception $e) {
    return false;
  }

  return true;
}
```

> This function calls another function that will need to be created which is responsible for fetching and caching the JWT keys from Okta, so create that function now:

Reviewer: ```suggestion
This function calls another function that you need to create that is responsible for fetching and caching the JWT keys from Okta. Create that function now:
```

Author: fixed
